### PR TITLE
:book Developer documentation

### LINKED ISSUE #778: Adding a development guide for cluster-api
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

**What did you expect to happen:**
It is better have a document to clarify some steps for how to do some development for this repo, this can help attract more contributors.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by liztio**
**What this PR does / why we need it**:
We have a bunch of empty holes in the book right now. This fills some of them in! 

Also I added a Makefile for rendering the images because it's a pain otherwise. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #778

### CODE REVIEW

File: docs/book/src/architecture/controllers/machine-set.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
 # MachineSet
-<!-- TODO -->
-This page is still being written - stay tuned!
+
+A MachineSet is an immutable abstraction over [Machines](./machine.html).
+
+Its main responsibilities are:
+* Adopting unowned Machines that aren't assigned to a MachineSet
+* Adopting unmanaged MachineSets that aren't assigned a Cluster
```
Comment (detiber) [inline docs/book/src/architecture/controllers/machine-set.md]:
```suggestion
* Adopting unmanaged Machines that aren't assigned a Cluster
```

**Comment by vincepri** [comment]
/assign @detiber 
/milestone v0.2.x

@liztio Feel free to change the milestone if this is going to be targeted only to v1alpha3

File: docs/book/src/images/Makefile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FILES := bootstrap-controller.plantuml
+FILES += cluster-admission-cluster-controller.plantuml
+FILES += cluster-admission-machine-controller.plantuml
+FILES +=cluster-admission-machineset-controller.plantuml
+FILES +=cluster-admission-machinedeployment-controller.plantuml
```
[the remark this answers is not in the document]
Comment (liztio) [reply docs/book/src/images/Makefile]: oh wow I didn't even notice that >_<

**Comment by detiber** [comment]
Updating the milestone, since this references the new location for the providers.

**Pull request merged**